getinput/getinput.go

```diff
@@ -31,8 +35,39 @@ func Get(year int, day int, sessionKey string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	return contents, nil
+}
+
+func Get(year int, day int, sessionKey string) ([]string, error) {
+	cacheFilename := filepath.Join(os.TempDir(), "advent", strconv.Itoa(day))
```

Comment:
Looks like you will need year also in the path:
`cacheFilename := filepath.Join(os.TempDir(), "advent", strconv.Itoa(year), strconv.Itoa(day))`